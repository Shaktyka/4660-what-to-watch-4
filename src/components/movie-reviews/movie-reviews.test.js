import React from 'react';

import renderer from 'react-test-renderer';
import NameSpace from "../../reducer/name-space";
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import MovieReviews from './movie-reviews.jsx';

const REVIEWS = [
  {
    id: 1,
    authorId: 4,
    authorName: `Kate Muir`,
    rating: 8.9,
    text: `Discerning travellers`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.DATA]: {
    films: [],
    filmReviews: [REVIEWS],
    isReviewsLoading: false,
  }
});

describe(`MovieReviews rendering`, () => {

  it(`MovieReviews renders correctly with the one review`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MovieReviews
              reviews={REVIEWS.slice(0, 1)}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
