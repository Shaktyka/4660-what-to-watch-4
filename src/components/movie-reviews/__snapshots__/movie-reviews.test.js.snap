// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieReviews rendering MovieReviews renders correctly with the one review 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
      id={1}
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2019-05-08T14:13:56.569Z"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
